src/poseidon.rs

```diff
@@ -75,7 +80,7 @@ impl<F: Field> MultiPermutation<F> for Poseidon<F> {
         assert!(self.num_rounds.full % 2 == 0, "asymmetric permutation configuration");
         let full_rounds_per_side = self.num_rounds.full / 2;
 
-        let mut current = inputs.to_owned();
+        let mut current = inputs.to_vec();//.to_owned();
```

Comment:
I think `to_vec()` is fine, let's delete `to_owned()`

Reply:
Agreed.